mantidproject/vesuvio
# Changing user experience

**Description of work:**
After meeting with Matthew, it was clear that a different way of interacting with the code was needed, as described in #104. The main changes this PR introduces are the following:

- Creation of new experiments is now done through manual copy-and-paste of a single `analysis_inputs.py` script
- Terminal functionality is still available but options have changed from `cache`, `experiment` and `ipfolder` to just `inputs` and `ipfolder`. This makes it harder to set up experiments through the terminal but makes it simpler to set up experiments when running an `analysis_inputs.py` script.

**To test:**



1. Create a new conda enviroment: `conda create -n mvesuvio-test`
2. Activate environment: `conda activate mvesuvio-test`
3. Install mantidworkbench (version needs to be 6.8): `mamba install mantidworkbench==6.8`
4. `cd` into the repository with this PR and install package by running `pip install -e .`
5. In the terminal, run the command `mvesuvio config` to set up default configuration
6. Go to the default folder, `~/.mvesuvio` and copy `analysis_inputs.py` file
7. Create a new directory anywhere in your machine and paste `analysis_inputs.py` file inside it.
8. Open Mantid workbench from the vesuvio conda environment (command `workbench`)
9. Inside workbench, open the `analysis_inputs.py` file you just pasted.
10. Run the file.
11. The file should run without issues and create a new folder with the same name as the file, i.e. check that a folder called `analysis_inputs` is created alongside the file `analysis_inputs.py`.
12. Check the folder contains the outputs of the inputs file.
13. Copy-and-paste the same `analysis_inputs.py` file in the same directory but this time rename it to something different
14. Run this file from inside workbench and check that a new directory with the new name is created.



Fixes #104.
